### ISSUE TITLE
Music is not getting ID's assigned

### ISSUE THREAD

**Issue by Duffyx** [MediaBrowser/Emby#3339]
I'd like to reopen the following issue, https://github.com/MediaBrowser/Emby/issues/3096.
We can continue discussing here though.

I've taken another shot at trying Emby. I'm determined to switch over (even bought premiere!).
I'm still dealing with this issue and have no clue why its happening with only my music collection.
I have also transitioned to the recommended library directory structure for music to be found on the wiki of Emby. Does emby even look at the directory structure to figure out album / artist?
I have no mp3 tags for my songs and my libraries are read/execute only (to emby). Altough don't think this is the issue as movies/series don't have this and are also read/execute only. I noticed that none of my albums have a music brains ID tag set (or any other). I'll readd my music library and attach a log later today. The reason why some of my albums where showing pictures is because they were embedded in the mp3 file. So they also didn't contain an ID for musicbrainz. I deleted all MP3 targ info in the files, even the picture and now nothing is getting musicbrain ID's and thus no pictures.

**Comment by Duffyx** [comment]
![image](https://user-images.githubusercontent.com/32487486/41818839-a02e6a2c-77b6-11e8-9df9-72ba6ddacee6.png)

[server-63665395200.txt](https://github.com/MediaBrowser/Emby/files/2130731/server-63665395200.txt)
[server-63665433074.txt](https://github.com/MediaBrowser/Emby/files/2130732/server-63665433074.txt)

Ok, so the picture to indicate what my library looks like atm.
Two log files attached when adding my music library.
No musing matching is happening. When entering ID targs of musicbrainz manually it works just fine.

**Comment by LukePulverenti** [comment]
Ah I see you're on FreeBSD, the version of Emby server is running a little behind at 3.3. The latest version is 3.4.1 but that hasn't been released for FreeBSD yet. 3.4.1 has some improvements for this, but the latest beta release has even more. We're soon going to begin development on our own FreeBSD package that we can make immediately available following every new release. You can read about that here:
https://emby.media/community/index.php?/topic/12906-freebsd-package/?p=588153

Some users in that topic have even figured out how to update their mono install manually and then manually update the server install, but we don't have any official instructions for that.

**Comment by Duffyx** [comment]
Alright thanks for the update Luke! Give FreeBSD some love! I guess we can leave this issue open. I'll get back to this once the new version lands.

**Comment by Duffyx** [comment]
Hello, I updated to emby package 3.5 for FreeBSD and readded my music library. Altough I'm still not seeing any pictures of metadata on the albums... :(

**Comment by Duffyx** [comment]
Ok provided a log file of adding my music library. Albums don't get musicbrainz ID's filled in automatically. I added my library at 10h:42m:50s. Do you think that adding mp3 tags to my music collection will help in this? Altough I prefer it fetches based on directory structure... a bit easier to maintain. I'm not sure if you can troubleshoot the problem from the log (debug logging enabled).

[Log.txt](https://github.com/MediaBrowser/Emby/files/2252878/Log.txt)

**Comment by Duffyx** [comment]
Ok so what I did as a test was the following; I gave the songs of an album mp3 tags (artist = ABBA, album = ABBA) and gave it a manual refresh. After the refresh it found musicbrainz ID's and filled them in. The given mp3 tags are identical to my directory structure i.e. :
/Artistname (ABBA)
- Albumname (ABBA)
-- Song1.mp3
-- Song2.mp3
-- Songx.mp3

So from mp3 tags it can obtain the album/artits and find the associated musicbrainz ID's.
